src/odb/src/db/dbBox.cpp

```diff
@@ -994,6 +994,34 @@ dbBox* dbBox::create(dbInst* inst_, int x1, int y1, int x2, int y2)
   return (dbBox*) box;
 }
 
+void dbBox::destroy(dbBox* box)
+{
+  _dbBox* db_box = (_dbBox*) box;
+  switch (db_box->_flags._owner_type) {
+    case dbBoxOwner::BPIN: {
+      _dbBPin* pin = (_dbBPin*) box->getBoxOwner();
+      pin->removeBox(db_box);
+      _dbBlock* block = (_dbBlock*) pin->getOwner();
+      block->_box_tbl->destroy(db_box);
```

Comment:
The ctor calls `block->add_rect(box->_shape._rect);` so this needs to call remove_rect.